### PR TITLE
Staff pick

### PR DESCRIPTION
**Pull request by liywjl**
### What?

When a user is of type staff, they should be able to see the `pick` button and are able to pick comments

Screenshot 2020-02-28 at 16 04 14

### Trello
https://trello.com/c/AauPx6Q2/1178-staff-can-pick-comments

### CODE REVIEW

File: src/components/Comment/Comment.tsx [hunk 1]
```diff
@@ -172,14 +200,24 @@ export const Comment = ({
...
-              <button className={commentControlsButtonStyles}>Pick</button>
+              {/* Only staff can pick, and they cannot pick thier own comment */}
+              {user &&
+                user.badge.some(e => e.name === "Staff") &&
+                user.userId !== scopedComment.userProfile.userId && (
+                  <button
+                    onClick={pick}
+                    className={commentControlsButtonStyles}
+                  >
+                    {scopedComment.isHighlighted ? "Unpick" : "Pick"}
```
Comment (oliverlloyd) [inline src/components/Comment/Comment.tsx]: If this PR isn't adding the ability to unpick a pick we shouldn't add the UI for that action. Or if we do it should be made clear that the logic is pending to help future developers / prevent it being missed. Maybe use the text: 'TODO: Unpick'

File: src/components/Comment/Comment.stories.tsx [hunk 2]
```diff
@@ -36,11 +36,36 @@ const commentData: CommentType = {
...
+export const Default = () => (
+  <Comment
+    comment={commentData}
+    pillar={"sport"}
+    setCommentBeingRepliedTo={() => {}}
+  />
+);
+Default.story = { name: "default" };
+
+export const StaffUser = () => (
```
Comment (oliverlloyd) [inline src/components/Comment/Comment.stories.tsx]: Thanks for the story!

File: src/components/Comment/Comment.tsx [hunk 3]
```diff
@@ -116,15 +118,41 @@ const Row = ({ children }: { children: JSX.Element | JSX.Element[] }) => (
 export const Comment = ({
   comment,
   pillar,
-  setCommentBeingRepliedTo
+  setCommentBeingRepliedTo,
+  user
 }: Props) => {
   const commentControlsButtonStyles = commentControlsButton(pillar);
+  const [scopedComment, setScopedComment] = useState(comment);
```
Comment (oliverlloyd) [inline src/components/Comment/Comment.tsx]: I don't think we should be loading the entire comment into state. This adds a lot of complexity to this component for a very limited benefit. It gives internal staff users immediate feedback on their actions but makes the codebase more complicated to reason about and increases the risk of bugs. Frontend doesn't have this feature - you only see updates after you refresh - so we have no requirement to actually add it

File: src/lib/api.tsx [hunk 4]
```diff
@@ -152,12 +155,9 @@ export const getCommentCount = (
...
 };
 
-export {
-  getDiscussion,
-  getProfile,
-  preview,
-  comment,
-  reply,
-  recommend,
-  getPicks
```
[the remark this answers is not in the document]
Comment (liywjl) [reply src/lib/api.tsx]: Good point, I will make it in a seperate PR. The issue I had we we were mixing `export const xxx` and 
```
export {
   xxx,
}
```

but granted, this should be outside the scope of this PR

File: src/components/Comment/Comment.tsx (hunk 1 above)
Comment (liywjl) [reply src/components/Comment/Comment.tsx]: great catch, super easy change to implement both

File: src/components/Comment/Comment.tsx [hunk 5]
```diff
@@ -119,11 +121,46 @@ export const Comment = ({
   comment,
   pillar,
   setCommentBeingRepliedTo,
+  user,
   isReply
 }: Props) => {
   const commentControlsButtonStyles = commentControlsButton(pillar);
+  const [isHighlighted, setIsHighlighted] = useState<boolean>(
+    comment.isHighlighted
+  );
```
Comment (oliverlloyd) [inline src/components/Comment/Comment.tsx]: Nice middle ground 👍

File: src/components/Comment/Comment.stories.tsx [hunk 6]
```diff
@@ -56,16 +56,39 @@ export const Default = () => (
...
+export const StaffUserUnpicked = () => (
+  <Comment
+    comment={commentData}
+    pillar={"sport"}
+    setCommentBeingRepliedTo={() => {}}
+    user={staffUser}
+    isReply={false}
+  />
+);
+StaffUserUnpicked.story = { name: "Staff user unpicked" };
```
Comment (oliverlloyd) [inline src/components/Comment/Comment.stories.tsx]: 💯

File: src/components/Comment/Comment.tsx [hunk 7]
```diff
@@ -151,46 +162,53 @@ export const Comment = ({
...
+                  >
+                    {comment.userProfile.displayName}
+                  </a>
                 </div>
                 <div className={timestampWrapperStyles}>
                   <Timestamp
-                    isoDateTime={scopedComment.isoDateTime}
-                    linkTo={`https://discussion.code.dev-theguardian.com/comment-permalink/${scopedComment.id}`}
+                    isoDateTime={comment.isoDateTime}
+                    linkTo={`https://discussion.code.dev-theguardian.com/comment-permalink/${comment.id}`}
```
Comment (oliverlloyd) [inline src/components/Comment/Comment.tsx]: Not an issue for this PR but we'll need to be careful where we've got hardcoded refs to both production and code existing at the same time. We'll need to think about either sticking with prod or abstracting the baseUrl

**Pull request merged**